build-system/babel-plugins/babel-plugin-jsx-style-object/index.js

```diff
@@ -0,0 +1,181 @@
+    const cssName = dashCase(name);
+    const isDimensional = !nonDimensional.has(cssName);
+    const value = path.get('value');
+    const evaluated = value.evaluate();
+
+    // If we can evaluate the value, return a composed string.
+    if (evaluated.confident) {
+      const {value} = evaluated;
+      if (value == null || value === '') {
+        return null;
+      }
+      const withUnit =
+        isDimensional && typeof value === 'number' ? `${value}px` : value;
+      return t.stringLiteral(`${cssName}:${withUnit};`);
+    }
+
+    // Otherwise call the helper function to evaluate nullish and dimensional values.
+    if (!hasHelperModule && programPath) {
+      hasHelperModule = true;
+      programPath.unshiftContainer(
+        'body',
+        t.importDeclaration(
+          [t.importSpecifier(t.identifier(helperFn), t.identifier(helperFn))],
+          t.stringLiteral(helperModule)
+        )
+      );
+    }
+    const args = [t.stringLiteral(cssName), value.node];
+    if (isDimensional) {
+      args.push(t.booleanLiteral(true));
+    }
+    return t.callExpression(t.identifier(helperFn), args);
+  }
+
+  /**
+   * @param {(babel.Node|null)[]} props
+   * @return {?babel.Node}
+   */
+  function mergeTransformedIntoExpr(props) {
+    let expr = null;
+    let prev = null;
+    while (props.length) {
+      const part = props.shift();
+      if (!part) {
+        continue;
+      }
+      // Collapse adjacent strings.
+      if (t.isStringLiteral(prev) && t.isStringLiteral(part)) {
+        // @ts-ignore
+        prev.value += part.value;
+      } else {
+        if (!expr) {
+          expr = part;
+        } else {
+          expr = t.binaryExpression('+', expr, part);
+        }
+        prev = part;
+      }
+    }
+    return expr;
+  }
+
+  return {
+    name: 'jsx-style-object',
+    visitor: {
+      Program: {
+        enter(path) {
+          programPath = path;
+          hasBaseModule = false;
+          hasHelperModule = false;
+          path.traverse({
+            ImportDeclaration(path) {
+              if (path.node.source.value.endsWith(helperModule)) {
+                hasHelperModule = true;
+              } else if (path.node.source.value.endsWith(baseModule)) {
+                hasBaseModule = true;
+              }
+              if (hasHelperModule && hasBaseModule) {
+                path.stop();
+              } else {
+                path.skip();
+              }
+            },
+          });
+        },
+      },
+      JSXAttribute(path) {
+        if (!hasBaseModule) {
+          return;
+        }
+        if (!t.isJSXIdentifier(path.node.name, {name: 'style'})) {
+          return;
+        }
+        const objectExpression = path.get('value.expression');
+        if (!objectExpression.isObjectExpression()) {
+          return;
+        }
+        const props = objectExpression.node.properties.map((_, i) => {
+          const prop = objectExpression.get(`properties.${i}`);
```

Comment:
You can iterate the paths directly
```suggestion
        const props = objectExpression.get('properties).map((prop) => {
```